src/components/ceremony/ceremony.edgedb.repository.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@nestjs/common';
+import { PublicOf } from '~/common';
+import { castToEnum, RepoFor } from '~/core/edgedb';
+import { CeremonyRepository } from './ceremony.repository';
+import { Ceremony, CeremonyType } from './dto';
+
+@Injectable()
+export class CeremonyEdgeDBRepository
+  extends RepoFor(Ceremony, {
+    hydrate: (ceremony) => ({
+      ...ceremony['*'],
+      engagement: true,
+      type: castToEnum(ceremony.__type__.name.slice(12, -8), CeremonyType),
+    }),
+  }).withDefaults()
+  implements PublicOf<CeremonyRepository>
+{
+  override create() {
```

Comment:
I think I have this mostly ready to go now with your other comments, but I do still have some errors here with the return types not matching. It feels like this is supposed to be the case as these are actually being created as part of the engagement schema, so not sure if this should just be ignored or if there is a true way to resolve the issue? @CarsonF 

TS2416: Property `create` in type `CeremonyEdgeDBRepository` is not assignable to the same property in base type `PublicOf<CeremonyRepository>`
Type () => void is not assignable to type
```
(input: CreateCeremony) => Promise<{   id: ID; }>
```
Type `void` is not assignable to type `Promise<{ id: ID; }>`
TS2416: Property create in type `CeremonyEdgeDBRepository `is not assignable to the same property in base type
```
RepoFor<typeof Ceremony, $expr_PathNode<TypeSet<$Ceremony, Cardinality.Many>, null>, { engagement: true; type: { [x: number]: $expr_Operator<ScalarType<"std::str", string, string, string>, Cardinality.One>; [x: `${number}:${number}`]: $expr_Operator<...>; [x: `${number}:`]: $expr_Operator<...>; [x: `:${number}`]: $e...
```
Type `() => void` is not assignable to type
```
(input: { id?: ID | TypeSet<$uuid, Cardinality.AtMostOne | Cardinality.One | Cardinality.Empty> | undefined; createdAt?: DateTime | TypeSet<...> | undefined;  ... 6 more ...; project: TypeSet<...> | ... 1 more ... | TypeSet<...>; } | ... 4 more ... | { ...; }) => Promise<...>
```
Type `void` is not assignable to type
```
Promise<Readonly<{   id: ID;   createdAt: DateTime;   sensitivity: 'Low' | 'Medium' | 'High';   type: 'Dedication' | 'Certification';   isMember: boolean;   engagement: Readonly<{     id: ID;   }>;   ... 4 more ...;   planned: boolean; }>>
```

Reply:
Yeah this is a tricky situation. In neo4j, the ceremony is created and then attached to the engagement. In EdgeDB, the engagement is given as a required link of the ceremony. This isn't the first time we've hit this.
For this case though, a schema trigger creates it for us, so we don't have to bother.
I'd just
```ts
create(input: CreateCeremony): never {
  throw new NotImplementedException();
}
```
That will make the types compatible, and we already the condition to not call any of this create logic for EdgeDB.

Reply:
Is this not done as an override? Still seeing TS having problems after pushing the most recent commit with this: 
```
  override create(input: CreateCeremony): never {
    throw new NotImplementedException();
  }
  ```
  
 `create` method throws this: 
 TS2416: Property `create` in type `CeremonyEdgeDBRepository` is not assignable to the same property in base type
```
RepoFor<typeof Ceremony, $expr_PathNode<TypeSet<$Ceremony, Cardinality.Many>, null>, { engagement: true; type: { [x: number]: $expr_Operator<ScalarType<"std::str", string, string, string>, Cardinality.One>; [x: `${number}:${number}`]: $expr_Operator<...>; [x: `${number}:`]: $expr_Operator<...>; [x: `:${number}`]: $e...
```
Type `(input: CreateCeremony) => never` is not assignable to type
```
(input: { modifiedAt?: DateTime | TypeSet<$datetime, Cardinality.AtMostOne | Cardinality.One | Cardinality.Empty> | undefined;  ... 7 more ...; engagement: TypeSet<...> | ... 1 more ... | TypeSet<...>; } | ... 4 more ... | { ...; }) => Promise<...>
```

Types of parameters `input` and `input` are incompatible.
Type
```
{ modifiedAt?: DateTime | TypeSet<$datetime, Cardinality.AtMostOne | Cardinality.One | Cardinality.Empty> | undefined;  ... 7 more ...; engagement: TypeSet<...> | ... 1 more ... | TypeSet<...>; } | ... 4 more ... | { ...; }
```
is not assignable to type `CreateCeremony`
Property `type` is missing in type
```
{ modifiedAt?: DateTime | TypeSet<$datetime, Cardinality.AtMostOne | Cardinality.One | Cardinality.Empty> | undefined;  ... 7 more ...; engagement: TypeSet<...> | ... 1 more ... | TypeSet<...>; }
```
but required in type `CreateCeremony`

Reply:
Ah that's unfortunate. So we will need to have this `create` in a `customize()`. Sorry for that whiplask.